test/test_jit.py

```diff
@@ -1384,5 +1387,20 @@ def to_int(x) -> (y):
         outputs = Variable(torch.IntTensor([1, 2]), requires_grad=True)
         self.checkScript(script, 'to_int', [x], [outputs], False)
 
+    @unittest.skipIf(PY2, "Python 2 not supported yet")
+    def test_python_frontend(self):
+        def fn(x, y, z):
+            q = x + y - z
+            w = -z
+            if not x and not y and z:
+                m = x if not z else y
+            while x < y > z:
+                q = x
+            return x
+
+        ast = torch.jit.python_frontend.get_jit_ast(fn)
+        self.assertExpected(str(ast))
+
+
```

Comment:
Can we add a test that source ranges are getting set correctly?

Reply:
Sure.